pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -133,6 +133,11 @@ public static String splitPath(String source, int slice) {
      * @return the web service caller identification
      */
     public String clientAppId() {
+        return "clientIP=" + httpRequest.getRemoteAddr() +
+                "clientRole=" + httpRequest.getAttribute(AuthenticationFilter.AuthenticatedRoleAttributeName);
+    }
+
+    public String clientRole() {
```

Comment:
It's better to change here, only changing the implementation of `clientAppId`. Then no need to replace the `clientRole` somewhere.

Reply:
thinks for review. i tried before. but after analysis the code, some code path use clientAppId to get authorization clientRole, direct change origin clientAppId may introduce other bug. sorry for change too much code but i think may be we can find a way to better make the change simle.